src/sparseml/onnx/utils/helpers.py

```diff
@@ -251,7 +251,11 @@ def extract_nodes_shapes_ort(model: ModelProto) -> Dict[str, List[List[int]]]:
 
     sess_options = onnxruntime.SessionOptions()
     sess_options.log_severity_level = 3
-    sess = onnxruntime.InferenceSession(model_copy.SerializeToString(), sess_options)
+    sess = onnxruntime.InferenceSession(
```

Comment:
```suggestion
    providers = (
        ["CUDAExecutionProvider", "CPUExecutionProvider"]
        if onnxruntime.get_device() == 'GPU'
        else ["CPUExecutionProvider"]
  
    sess = onnxruntime.InferenceSession(
```

Reply:
Added optional support for onnxruntime-gpu. The point is that the users who installed onnxruntime-gpu and are able to run ort on GPU, do so by default.